server/src/test/DocumentFormatRequestHandler.test.ts

```diff
@@ -0,0 +1,150 @@
+import { DEFAULT_DOC_FORMAT_SETTINGS, IDocumentFormatSettings } from '../settings/Settings';
+
+const FAKE_FILE_PATH = '/file1.kao';
+const YML_LANGUAGE_ID = 'yml';
+
+/**
+ * Create and return a TextDocument with `docContent` as content.
+ *
+ * @param docContent the document content
+ *
+ * @return a TextDocument
+ */
+function createFakeDocument(docContent: string): TextDocument {
+    return TextDocument.create(FAKE_FILE_PATH, YML_LANGUAGE_ID, 1, docContent);
+}
+
+describe('DocumentFormatRequestHandler', () => {
+    it('should not give edits when there is nothing to do', (done) => {
+        const file = createFakeDocument(
+            `Person personA
+            --> lastName "LastName"
+            --> firstName "FirstName"
+            ;`,
+        );
+        // No edit required.
+        expect(buildDocumentEditList(file, DEFAULT_DOC_FORMAT_SETTINGS)).toHaveLength(0);
+        done();
+    });
+    it('should give edits when there are edits to do', (done) => {
+        const file = createFakeDocument(
+            `
+function myFunction(Object input)
+--> domains Object
+--> local Country country, Number a, Number b, Number x
+--> action {
+    /* Semicolon addition → “a = random();” */
+    a = random()
+    /* Semicolon addition → “b = random();” */
+    b = random()
+    /* Three space additions → “if(input.isInstanceOf(Country) != true) {” */
+    if(input.isInstanceOf(Country)!=true){
+        country = input;
+    }
+
+    /* One space removal + two space additions → “if(input.isInstanceOf(Number) == true) {” */
+    if (input.isInstanceOf(Number) ==true){
+        /* Semicolon addition → “country = dummyCountry.getCountry(input);” */
+        country = dummyCountry.getCountry(input)
+    }
+    /* One space removal + wo space additions → “if(input.isInstanceOf(Text) == false) {” */
+    if (input.isInstanceOf(Text)== false){
+        /* Semicolon addition → “country = dummyCountry.findCountryByCode(input);” */
+        country = dummyCountry.findCountryByCode(input)
+        /* Five space replacements → “}else if(a == b || a > b) {” */
+    }else if(a  ==        b    ||a     >   b) {
+        /* One space replacement + three space additions → “x = ifExpr(a > b)” */
+        x    =ifExpr(a>b)
+        then a
+        /* Semicolon addition → “else b;” */
+        else b
+        /* Semicolon addition → “logInfo(x);” */
+        logInfo(x)
+    /* One space replacement then two spaces replacements. */
+    /* “if(a <   b
+    ||  b    > x)”
+
+    → “if(a < b
+    || b > x)”
+        */
+    } else if(a <   b
+    ||  b    > x) {
+        hashMap[country] = a;
+    }
+    /* Two space additions + semicolon addition → “hashMap[country] = x;” */
+    hashMap[country]=x
+    /* Semicolon addition → “assert(a == b);” */
+    assert(a == b)
+    /* Semicolon addition → “return country;” */
+    return country
+};
+`,
+        );
+        expect(buildDocumentEditList(file, DEFAULT_DOC_FORMAT_SETTINGS)).toHaveLength(33);
```

Comment:
I expected to find a comparison with the full resulting document :sweat_smile:

Reply:
Couldn't do that. I wanted to do so but couldn't. :sob:

Reply:
Then, as I said, it might be more useful and maintainable to split examples if you can. Parsing a huge chunk of code and then just checking that there were exactly 287562876 suggested changes sounds kinda weird. xD

Reply:
If the test breaks because the number of changes went from 287562856 to 287562855, you may have a hard time finding out what exactly went wrong xd

Reply:
You're right. I'll do the changes.

Reply:
Sounds rather good. I’ll try to take another look at the diff before too long.